[ETC] Fixing bugs and refactoring

AndreyLykhoman opened the pull request:
**Description:** Refactoring. Adding checks to the code. 

**Youtrack:** https://youtrack.raccoongang.com/issue/ETC-11

idegtiarov commented on lms/djangoapps/etc_api/views.py:
```diff
@@ -37,36 +38,41 @@ def post(self, request):
             )
 
+        data['name'] = prename
+        data['name'] =  "{} {}".format(data['name'], surname) if data['name'] and surname else data['name']
```
I would use `strip` method:
"{} {}".format(prename, surname).strip() if prename or surname else username

idegtiarov commented on lms/djangoapps/etc_api/views.py:
```diff
@@ -37,36 +38,41 @@ def post(self, request):
                 status=400
             )
 
+        data['name'] = prename
```
This line could be omitted

idegtiarov commented on lms/djangoapps/etc_api/views.py:
```diff
@@ -37,36 +38,41 @@ def post(self, request):
 
+        data['name'] = prename
+        data['name'] =  "{} {}".format(data['name'], surname) if data['name'] and surname else data['name']
+        data['name'] = data['name'] if data['name'] else username
```
ditto

idegtiarov commented on lms/djangoapps/etc_api/views.py:
```diff
@@ -93,12 +99,16 @@ def post(self, request):
-            user = User.objects.get(id=data['user_id'])
-            user.is_active = string_to_boolean(data['is_active'])
+            user = User.objects.get(id=user_id)
+            user.is_active = string_to_boolean(data.get('is_active'))
```
is_active expects boolean please add the check

AndreyLykhoman commented:
@idegtiarov 
I rewrote code that related with creating 'name' field. I added default value as ''  to 'prename' and 'surname' fields. This is necessary for a new construction. It helps to dismiss 'None' string from the 'name' field.

idegtiarov commented on lms/djangoapps/etc_api/views.py:
```diff
@@ -116,17 +125,30 @@ def post(self, request):
+        except InvalidKeyError:
+            return Response(data={"user_message": "Wrong 'course_id'. Course does not exist"}, status=400)
+
+        return Response(data={'enrollment_id': enrollment_obj.id}, status=status.HTTP_200_OK)
```
If you use `HTTP_200_OK` why you don't use `HTTP_400_BAD_REQUEST` in lines L152 and 139

AndreyLykhoman replied:
I fixed that mistakes.
